cmd/snap-confine/snap-confine.apparmor.in

```diff
@@ -38,6 +38,8 @@
     /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libudev.so* mr,
     /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libseccomp.so* mr,
     /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libcap.so* mr,
+    # Needed to run /usr/bin/sh for snap-device-helper.
+    /{,usr/}lib{,32,64,x32}/{,@{multiarch}/}libtinfo.so* mr,
```

Comment:
This rule is fine from a security POV, but I wouldn't normally be super excited about it from a maintenance POV, but we know that snap-confine is going to be adjusted to not shell out to snap-device-helper, so this is ok for now (at such time, we should remove the access that only the shell needs, like this one, but I suspect there are more).